sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -427,7 +424,14 @@ def post_port_dom_info_to_db(logical_port_name, table, stop_event=threading.Even
         ganged_member_num += 1
 
         try:
-            dom_info_dict = _wrapper_get_transceiver_dom_info(physical_port)
+            if dom_info_cache is not None and physical_port in dom_info_cache:
+                # If cache is enabled and dom information is in cache, just read from cache, no need read from EEPROM
+                dom_info_dict = dom_info_cache[physical_port]
+            else:
+                dom_info_dict = _wrapper_get_transceiver_dom_info(physical_port)
+                if dom_info_cache is not None:
+                    # If cache is enabled, put dom information to cache
+                    dom_info_cache[physical_port] = dom_info_dict
```

Comment:
This is for performance. Multiple logical port might belong to same physical port, in that case, we only need read the EEPROM once and cache EEPROM value for other logical port belongs to the same physical port. Reading EEPROM is usually a long call because it accesses hardware.